### PR TITLE
Update unit tests

### PR DESCRIPTION
**Pull request by drake-vcu**
- [x] I have read the Code of Conduct and the Contributing files before opening this issue.
- [x] I have verified the issue this pull request fixes or feature this pull request implements is to the current release.
- [x] I am using the latest stable release with the above code fix or the current master branch if code changes are present (prerelease code changes).
- [x] I have debugged my changes before submitting this issue and that everything should just work.

# Describe the issue this fixes / feature this adds

I have made small fixes to the main class, each denoted by individual commits

# Test Results

Test results are all successful, I was able to find some bugs along the way.
Because some of the function here such as "Read" also do an Add, it is difficult to isolate some of the testing.

### CODE REVIEW

File: XmlAbstraction/XmlObject.cs [hunk 1]
```diff
@@ -656,6 +650,8 @@ public void Delete(string elementname, string attributename)
                 {
                     throw new ArgumentException("elementname or attributename does not exist in the xml or in pending edits.");
                 }
+
+                this.HasChanged = true;
```
Comment (AraHaan) [inline XmlAbstraction/XmlObject.cs]: Should we only set this to true only if it is not then set it to true?

**Comment by AraHaan** [comment]
The read functions fallsback to a Add only if the elements or attributes being read do not exist. Although I am not sure if it should throw if they do not exist and force the user to add them or just do nothing and return an empty collection of strings (or a single empty string if it returns only a single string).

File: XmlAbstraction/XmlObject.cs (hunk 1 above)
Comment (drake-vcu) [reply XmlAbstraction/XmlObject.cs]: Its not really a big deal considering this is just a bool and the actual setting of it doesn't affect anything in the code. But if you want to you could do it in the property setter, but consequently will have to declare the underlying variable and not use auto property.

Comment (AraHaan) [reply XmlAbstraction/XmlObject.cs]: Wow I had to manually rebuild the PR on the CI project page in order to build the latest commit.

Comment (AraHaan) [reply XmlAbstraction/XmlObject.cs]: Now that I think about it we should increment the version to 1.1.2.

**Comment by drake-vcu** [comment]
> 
> 
> Seems on the report a few tests are failing (ones that is expected to pass on some calls) on the CI and seems ArgumentException is not thrown in the tests or probably tested for.

Some of those tests depend on a specific drive location to exist, they may need to be rewritten to be drive and platform independent.

**Comment by AraHaan** [comment]
ye, I know, Also the test that expects a exception but fails because the exception does not throw could be because the tests are running in administrator mode and gives access to the drive root?

**Comment by drake-vcu** [comment]
you are probably right. should reconstruct test to save somewhere that was explicitly set to restricted for testing.

**Comment by AraHaan** [comment]
hmm try the tests in admin mode on your machine but point it to places like "%WinDir%\SoftwareDistribution" (aka "%SystemDrive%\Windows\SoftwareDistribution") that Windows Update downloads updates to.

**Pull request merged**